### ISSUE TITLE
[Client bug] Microsoft.Graph 4.1.0 fails in UWP apps build with .NET Native [Regression]

### ISSUE THREAD

**Issue by mfeingol** [microsoftgraph/msgraph-sdk-dotnet#1090]
**Describe the bug**
Previously working UWP apps stop working after upgrading to Microsoft.Graph 4.0.0 or 4.1.0.

Concretely, operations such as Microsoft.Graph.DriveItemRequest.GetAsync() throw exceptions:

Microsoft.Graph.ServiceException: Code: generalException Message: Unexpected exception returned from the service. ClientRequestId: [guid]

**To Reproduce**
1. Have a UWP app that uses Microsoft.Graph to connect to OneDrive
2. Upgrade Microsoft.Graph NuGet package to 4.1.0
3. Compile with .NET Native
4. Observe exceptions thrown from OneDrive operations such as the one mentioned above.

**Expected behavior**
No regressions upgrading from 3.35.0 to 4.1.0

**Desktop (please complete the following information):**
 - OS: Windows 10 19043

**Additional context**

I will provide more information, including a repro project, as soon as I can.

Odds are this can be worked around using an rd.xml file. However, that has other issues (increases build size, difficult to discover for developers, can cause build failures, etc.). Further, this is a regression from Graph 3.35.0. 
[AB#10568](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/10568)

**Comment by mfeingol** [comment]
Here's a repro project:

[GraphRepro.zip](https://github.com/microsoftgraph/msgraph-sdk-dotnet/files/6949622/GraphRepro.zip)

To make the repro app work, you'll need to register an app as per https://aka.ms/msal-net-register-app and update main.xaml.cs with your app's client id.

You'll also need to enable .NET Native compilation on the project to see the bug. When you build with .NET Native, you'll see that the DriveItem obtained from `Me.Drive.Root` has all null fields. This is the bug.

If you enable exclusions for Microsoft.Graph in the rd.xml file, the program will work.

As noted, this is a very bad workaround because it's difficult to discover, it impacts compilation times and sizes, and it is a regression from previous versions of the library.

I'll note that in my particular app, enabling this workaround causes .NET Native builds to fail with the following error: `error : RHBIND : error RHB0002: Failed to write PDB.` This is a symptom of the build getting too large for the 32-bit rhbind to handle inside its address space, and it means that without a proper fix in the Graph library itself that makes this unnecessary, I will be unable to upgrade to Microsoft.Graph 4.0.

**Comment by andrueastman** [comment]
Hey @mfeingol,

Thanks for raising this issue and bringing it to our attention. 

Are you able to share the sample rd.xml with the exclusions that make the repro app work?

**Comment by mfeingol** [comment]
@andrueastman: yep, if you uncomment either of these, the repro app works:

```xml
  <Application>
    <!--
      An Assembly element with Name="*Application*" applies to all assemblies in
      the application package. The asterisks are not wildcards.
    -->
    <!--<Assembly Name="*Application*" Dynamic="Required All" />-->
    
    
    <!-- Add your application specific runtime directives here. -->
    <!--<Assembly Name="Microsoft.Graph" Dynamic="Required All" />-->

  </Application>
```

**Comment by 27k1** [comment]
I have the same issue exactly and it means I cannot deploy in native code. The same issue is in Visual Studio 2022 Preview, it is supposed to be 64bit yet 32-bit rhbind has not been updated to 64bit. This wastes so many hours trying to deploy UWP apps in native mode and 64bit bind is really urgent.

**Comment by 27k1** [comment]
The solution is to remove the app centre crash reporting and insert this new tag in the project file.
 <NoLinkerSymbols>true</NoLinkerSymbols>
Then use a suitable logging method such as SeriLog to do your crash reporting, perhaps with the SeriLog Azure storage API.

The .NET native toolchain dev team is not part of the Visual Studio dev team so I have no hope that this native tool chain will ever be developed as 64bit.

**Comment by tipa** [comment]
Seems like with 4.4.0 there has been a a [rd.xml file added](https://github.com/microsoftgraph/msgraph-sdk-dotnet/commit/d92dc7b2660aedb6d6b64af261fd8e58c1114dbf) and the problem is gone.
Unfortunately, this again increases my apps size **by 10MB** (1/5th of it's total size!)

Perhaps I would have preferred to manually adjust the Linking, which I already do in my Xamarin applications with the LinkDescription.xml file

**Comment by andrueastman** [comment]
Thanks for the tip @tipa. 

From my testing, the package was already significantly smaller with the suggested [rd.xml](https://github.com/microsoftgraph/msgraph-sdk-dotnet/issues/1090#issuecomment-896139591) defined in the Graph library than when defined in the user project.

Furthermore, just as you observed, this could be reduced further with something similar to this in the user project if the package size is an issue.

```xml
    <Namespace Name="Microsoft.Graph" Serialize="Excluded" />
    <Type Name="Microsoft.Graph.DriveItem" Dynamic="Required All" />
```

The big win is that users can work with the package with UWP out of the box and optimize further if they please.
Closing this for now as this is resolved with 4.4.0 release.
